test/applications_integration_test.go

```diff
@@ -786,7 +786,9 @@ func theContentsOfTheBoxWithNameShouldBeIfThereIsAnErrorItIs(fromClient, encoded
 		err = fmt.Errorf("expecting algod or indexer, got " + fromClient)
 	}
 	if err != nil {
-		if strings.Contains(err.Error(), errStr) {
+		// If the expected error string is not empty, check if it is a substring of the actual error string.
+		// Note that if the expected error string is empty, then the second condition will always return true.
+		if len(errStr) != 0 && strings.Contains(err.Error(), errStr) {
 			return nil
 		}
 		return err
```

Comment:
This improvement solves the problem at hand, but we should also fail if we expected an error and none was reported. Something like this:

```go
if err != nil {
  // existing if statement body, guaranteed to return
}
if len(errStr) != 0 {
  return errors.New("expected an error but none was reported")
}
```

Reply:
Agreed, good point - added check for this in [7d6e2a1](https://github.com/algorand/go-algorand-sdk/pull/447/commits/7d6e2a1c3f04390f7393966c284fd2af8aa3666c)